Guide: Import a Bearer Did

blackgirlbytes opened the pull request:
https://deploy-preview-1628--tbd-website-developer.netlify.app/docs/web5/decentralized-identifiers/how-to-reuse-a-did

This pull request introduces a new guide for importing Decentralized Identifiers (DIDs) and includes corresponding test cases in JavaScript, Kotlin, and Swift. The guide explains how to export and import DIDs, ensuring users can maintain their identity across different platforms.

### Documentation:

* **New Guide for Importing DIDs**: Added a detailed guide on how to export and import DIDs, including code snippets for JavaScript, Kotlin, and Swift. (`site/docs/web5/decentralized-identifiers/how-to-import-did.mdx`)

### Test Cases:

* **JavaScript Tests**: Added tests to verify the export and import functionality of DIDs using the `DidDht` and `DidJwk` methods. (`site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-import-did.test.js`)
* **Kotlin Tests**: Implemented test cases for exporting and importing DIDs using `DidDht`, `DidJwk`, and `BearerDid` classes. (`site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToImportDidTest.kt`)
* **Swift Tests**: Added tests to ensure the correct export and import of `DIDJWK` using the `InMemoryKeyManager`. (`site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/HowToImportDidTests.swift`)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205459115459785
  - https://app.asana.com/0/0/1208314295097366

acekyd commented:
Nice! I was planning a short to cover this as it came up during the hackathon - but I ran into a question I want to pose now. For the title, would it be better to use something like "Managing DIDs or reusing DIDs etc since it covers both exporting and importing - for the sole purpose of reusing an existing DID 🤔

blackgirlbytes commented:
> Nice! I was planning a short to cover this as it came up during the hackathon - but I ran into a question I want to pose now. For the title, would it be better to use something like "Managing DIDs or reusing DIDs etc since it covers both exporting and importing - for the sole purpose of reusing an existing DID 🤔

oh okay! That's fair. Angie assigned a ticket to me called Import DIDs so I just named it that LOL but that's fair/a clearer name

angiejones commented on site/docs/web5/decentralized-identifiers/how-to-import-did.mdx:
```diff
@@ -0,0 +1,40 @@
+In this guide, you will learn how to take your [Decentralized Identifier (DID)](/docs/web5/decentralized-identifiers/what-are-dids) with you to other apps, allowing you to keep your identity consistent everywhere across various platforms. 
+
+## Export DID
+You will start by exporting your DID. This method converts your [Bearer DID](/docs/web5/decentralized-identifiers/how-to-create-did/#create-did) into a [Portable DID](/docs/web5/decentralized-identifiers/how-to-create-did/#portable-did), a serialized and static representation suitable for storage or transfer. 
```
we should add glossary terms for Bearer DID and Portable DID and link to those

blackgirlbytes replied:
added!

angiejones commented on site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-import-did.test.js:
```diff
@@ -0,0 +1,36 @@
+  
+      // :snippet-start: exportDidJs
+      // export did:dht DID
+      const portableDhtDid = await didDht.export();
```
in these 2 lines, it's not clear that didDht is a bearer did. can you rename these variables up top to `dhtBearerDid` and `jwkBearerDid`?

blackgirlbytes replied:
done

angiejones commented on site/docs/web5/decentralized-identifiers/how-to-import-did.mdx:
```diff
@@ -0,0 +1,40 @@
+
+
+## Import DID
+Now, you can import your Portable DID to a desired location using the `import` method. 
```
```suggestion
You can import your Portable DID to an application using the `import` method. 
```

angiejones commented on site/docs/web5/decentralized-identifiers/how-to-import-did.mdx:
```diff
@@ -0,0 +1,40 @@
+In this guide, you will learn how to take your [Decentralized Identifier (DID)](/docs/web5/decentralized-identifiers/what-are-dids) with you to other apps, allowing you to keep your identity consistent everywhere across various platforms. 
+
+## Export DID
+You will start by exporting your DID. This method converts your [Bearer DID](/docs/web5/decentralized-identifiers/how-to-create-did/#create-did) into a [Portable DID](/docs/web5/decentralized-identifiers/how-to-create-did/#portable-did), a serialized and static representation suitable for storage or transfer. 
```
can you specify what exactly gets returned? is it a JSON?

blackgirlbytes commented:
## Updates

- Changed the title from Import a DID to Reuse a DID based on @acekyd's suggestions
- Added Bearer DID and Portable DID to glossary, updated mentions of it based on @angiejones's suggestions.
